components/suggest/src/db.rs

```diff
@@ -128,6 +129,48 @@ impl<'a> SuggestDao<'a> {
         Self { conn, scope }
     }
 
+    // =============== High level API ===============
+    //
+    //  These methods combine several low-level calls into one logical operation.
+
+    pub fn handle_unparsable_record(&mut self, record: &RemoteSettingsRecord) -> Result<()> {
+        let record_id = SuggestRecordId::from(&record.id);
+        // Remember this record's ID so that we will try again later
+        self.put_unparsable_record_id(&record_id)?;
+        // Advance the last fetch time, so that we can resume
+        // fetching after this record if we're interrupted.
+        self.put_last_ingest_if_newer(record.last_modified)
+    }
+
+    pub fn handle_ingested_record(&mut self, record: &RemoteSettingsRecord) -> Result<()> {
+        let record_id = SuggestRecordId::from(&record.id);
+        // Remove this record's ID from the list of unparsable
+        // records, since we understand it now.
+        self.drop_unparsable_record_id(&record_id)?;
+        // Advance the last fetch time, so that we can resume
+        // fetching after this record if we're interrupted.
+        self.put_last_ingest_if_newer(record.last_modified)
+    }
+
+    pub fn handle_deleted_record(&mut self, record: &RemoteSettingsRecord) -> Result<()> {
+        let record_id = SuggestRecordId::from(&record.id);
+        // Drop either the icon or suggestions, records only contain one or the other
```

Comment:
This is my assumption, can someone verify if it's true or not?

Reply:
Yep, that's correct!